x-pack/filebeat/docs/inputs/input-aws-s3.asciidoc

```diff
@@ -115,6 +115,43 @@ it was uploaded. For example: `application/json`.
 The file encoding to use for reading data that contains international
 characters. This only applies to non-JSON logs. See <<_encoding_3>>.
 
+[id="input-{type}-decoding"]
+[float]
+==== `decoding`
+
+The file decoding option is used to specify a codec that will be used to
+decode the file contents. This can apply to any file stream data.  
+An example config is shown below:
+
+[source,yaml]
+----
+  decoding.codec.parquet: enabled
+----
+
+Currently supported codecs are given below:-
+
+    1. <<attrib-decoding-parquet,Parquet>>: This codec decodes parquet compressed data streams.
+ 
+[id="attrib-decoding-parquet"]
+[float]
+==== `the parquet codec`
+The `parquet` codec is used to decode parquet compressed data streams.
+Only enabling the codec will use the default codec options. The parquet codec supports
+two sub attributes which can make parquet decoding more efficient. The `batch_size` attribute and 
+the `process_parallel` attribute. The `batch_size` attribute can be used to specify the number of
+records to read from the parquet stream at a time. By default the `batch size` is set to `1` and 
+`process_parallel` is set to `false`. If the `process_parallel` attribute is set to `true` then functions 
+which read multiple columns will read those columns in parallel from the parquet stream with a 
+number of readers equal to the number of columns. Setting `process_parallel` to `true` will greatly
+increase the rate of processing at the cost of increased memory usage. Having a larger `batch_size`
+also helps to increase the rate of processing. An example config is shown below:
+
+[source,yaml]
+----
+  decoding.codec.parquet: enabled
+  decoding.codec.parquet.process_parallel = true
+  decoding.codec.parquet.batch_size = 1000
```

Comment:
I was surprised to find out that we can use config this way (`decoding.codec.parquet: enabled`).

Since we define the struct as:

```golang
type parquetCodecConfig struct {
	Enabled         bool `config:"enabled"`
	ProcessParallel bool `config:"process_parallel"`
	BatchSize       int  `config:"batch_size" default:"1"`
}
```

I would expect something along the line of:

```yaml
decoding.codec.parquet.enabled = true
...
```

Good to know.

Reply:
actually all of them should ideally be ":" and not "=" since the config is an yaml, thanks for the review, will need to update the docs.

Reply:
Yeah, I mindlessly followed the example below, my bad.

However, I was surprised by `decoding.codec.parquet: enabled` vs. `decoding.codec.parquet.enabled: true`.

Thanks for confirming.

Reply:
yea it should be decoding.codec.parquet.enabled: true not decoding.codec.parquet: enabled, it was a typo in the docs, updated it now.